planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h

```diff
@@ -336,8 +336,19 @@ class PlanningSceneMonitor : private boost::noncopyable
   /** @brief This function is called every time there is a change to the planning scene */
   void triggerSceneUpdateEvent(SceneUpdateType update_type);
 
-  /** \brief Wait until all pending scene updates with timestamps < t are incorporated */
-  void syncSceneUpdates(const ros::Time &t = ros::Time::now());
+  /** \brief Wait for robot state to become more recent than t.
+   *
+   * If there is no state monitor active, there will be no scene updates.
+   * Hence, you can specify a timeout to wait for those updates. Default is 1s.
+   */
+  bool waitForCurrentRobotState(const ros::Time &t, double wait_time = 1.);
+
+  /** \brief Wait current robot state and wait for all pending scene updates to be processed.
+   *
+   * Additionally to waitForCurrentRobotState() this processes also all already received scene updates
+   * as long as wait_time is not exceeded.
+   */
+  bool syncSceneUpdates(const ros::Time &t = ros::Time::now(), double wait_time = 1.);
```

Comment:
-1 But you knew that before, because we disagreed on that method before. :)
The problem here is that PlanningScenes do not know the concept of "submission time" and there is, even in theory, no way to make sure we updated the scene "to the current state", because packages could still linger about in the network and this function will miss these updates.
Thus,

* the function does not hold what its name promises (I know at least 5 students from my lab who wouldn't care about the "already received" part in the comment)
* users still have to make sure their updates arrived using an external mechanisms (many users prefer `sleep(5.0)`)
* Many users would surely call this function instead of making sure things are alright on their own and during their next demo the Heisenbug strikes and they miss their decisive scene update.

@davetcoleman , @mikeferguson I suppose we need a third opinion on whether we should have that function in the API.

I consider it to be too misleading and broken by design, @rhaschke considers it to be a reasonable addition.